### PR TITLE
ref(control_silo): Consolidate integrations api endpoints and tests

### PR DESCRIPTION
**Pull request by Christinarlong**
Moves all integrations related endpoints to new space sentry/integrations/api/endpoints and updates any import paths. In addition to any integrations endpoint tests. Includes import shims for modules that are imported in getsentry

ref(#73859)

### CODE REVIEW

File: src/sentry/api/endpoints/codeowners/external_actor/team_details.py [hunk 1]
```diff
@@ -1,81 +1,5 @@
-import logging
-from typing import Any
+from sentry.integrations.api.endpoints.codeowners.external_actor.team_details import (
```
Comment (markstory) [inline src/sentry/api/endpoints/codeowners/external_actor/team_details.py:1]: Do we need all these directories? We could flatten this to `sentry.integrations.api.endpoints.codeowners` and rename endpoint module to `external_actor_team_details`.

File: src/sentry/api/urls.py [hunk 2]
```diff
@@ -113,6 +112,57 @@
...
+    OrganizationIntegrationDetailsEndpoint,
+    OrganizationIntegrationRequestEndpoint,
+    OrganizationIntegrationsEndpoint,
+    OrganizationPluginsConfigsEndpoint,
+    OrganizationPluginsEndpoint,
+)
+from sentry.integrations.api.endpoints.internal.integration_proxy import (
+    InternalIntegrationProxyEndpoint,
+)
+from sentry.integrations.api.endpoints.organization.organization_code_mapping_codeowners import (
```
Comment (markstory) [inline src/sentry/api/urls.py:136]: These endpoints could be moved up to `sentry.integrations.api.endpoints` There aren't *that* many endpoints within integrations yet.

File: src/sentry/api/endpoints/codeowners/external_actor/team_details.py (hunk 1 above)
Comment (Christinarlong) [reply src/sentry/api/endpoints/codeowners/external_actor/team_details.py:1]: 👍 that sounds cleaner

**Review by markstory — APPROVED** [review]
Looking good :clap:

**Pull request merged**